### PR TITLE
[#9079] Separate component tests and E2E tests as clear as possible

### LINKED ISSUE #9079: Set up foundation for front-end migration project
To kick off [this project](https://github.com/TEAMMATES/teammates/projects/4), some foundations need to be laid out first:
- [x] Step 1: Set up Angular 6 front-end project
  - Included: lint/build/test routine (including reliable code coverage), separate dev server from the back end, hot reload in development mode, source maps, minification when building for production usage
- [x] Step 2: Set up REST-ful back end servlet, ~possibly with the help of [Jersey](https://medium.com/@tejohnso/a-minimal-java-api-with-jetty-jersey-jackson-and-gradle-9842a145815e),~ and user authentication workflow
- [x] Step 3: Migrate few pages; concurrently, set up example of how migration for each page should be done systematically.

- [ ] Step 4 (can be deferred): Set up example tests, ranging from front-end component tests (with Karma/Jasmine), API endpoint tests (with TestNG), and E2E tests (with Selenium).

### PR DESCRIPTION
**Pull request by wkurniawan07**
Step 1.5 of #9079. 

Changes:

- Made the separation of component tests and E2E tests as clear as possible; different `TestProperties`, different TestNG XML file, etc.
- Moved classes that are only used for E2E tests (e.g. `HtmlHelper`, `GmailServiceMaker`, `EmailAccount`, `BackDoor`) to `e2e` package.
- Excluded all test code from code coverage report.
- Updated testing instructions.

### CODE REVIEW

File: docs/development.md [hunk 1]
```diff
@@ -202,80 +198,67 @@ Assuming you have already done the UI-related change and tested it manually, the
...
+
+To run all front-end component tests once and generate coverage data afterwards, run the following command:
+```sh
+npm run coverage
+```
+
+To run an individual test or test group, change `it` or `describe` in any `*.spec.ts` file to `fit` or `fdescribe` respectively.
 
-Before running any test suite, it is important to have the dev server running locally first if you are testing against it.
+#### Running the tests with command line
```
[the remark this answers is not in the document]
Comment (xpdavid) [reply docs/development.md]: And then another `Running the e2e tests`

In each section, different running methods may be presented (with command line, IDE etc.)

[the remark this answers is not in the document]
Comment (xpdavid) [reply docs/development.md]: Or we should not organize the content based on kinds of test but running methods.

[the remark this answers is not in the document]
Comment (wkurniawan07) [reply docs/development.md]: I changed to organizing by running method; only marginally better IMO but still better than nothing.

**Comment by wkurniawan07** [comment]
> As the testng xml changed, we need update Component_Tests.xml E2E_Tests.xml for IntelliJ, eclipseLaunches/*.launch.xml also.

Already done in 17f2b98 before.

**Comment by xpdavid** [comment]
> Already done in 17f2b98 before.

I mean the path to the xml file. 

e.g. for e2e test, the testng xml file is in `$PROJECT_DIR$/src/e2e/resources/testng-e2e.xml` rather than `$PROJECT_DIR$/src/test/testng-e2e.xml` right now. Similar to component test also.

**Pull request merged**